index.spec.js

```diff
@@ -1,6 +1,7 @@
 'use strict'
 
-const Joi = require('@hapi/joi').extend(require('.'))
+const { semver, semverRange } = require('.')
+const Joi = require('@hapi/joi').extend(semver).extend(semverRange)
```

Comment:
Does this mean the two rules will need to be imported separately? Would it be possible to use an API like `Joi.semver().range().valid()` instead of `Joi.semverRange().valid()` allow this to be imported as a single extension?

Reply:
if I understood correctly, you cannot chain `range` after `semver` as a range is not necessarily a valid version.

Reply:
> Does this mean the two rules will need to be imported separately?

Yes - this would change the documented import/extend syntax to

```js
const { semver } = require('joi-extension-semver')
const Joi = require('@hapi/joi').extend(semver)
```

if you only care about a version, and then 

```js
const { semverRange } = require('joi-extension-semver')
const Joi = require('@hapi/joi').extend(semverRange)
```

if you want to validate a range.


> Would it be possible to use an API like `Joi.semver().range().valid()`...

> if I understood correctly, you cannot chain range after semver as a range is not necessarily a valid version.

Yes - exactly. In order to do `Joi.semver().range().valid()`, your value has to pass `Joi.semver().valid()` first and `>=5.0.1` is a valid range, but it isn't a valid version, for example, so it never gets that far because it gets checked as a version first.